### DISCUSSION TITLE
Filtering via facettes instead of fix tree/hierarchy

### DISCUSSION THREAD

**Discussion by AtrusRiven** [squidfunk/mkdocs-material#6282] [Ideas]
There are cases where you do not want your documentation to be organized via a static/fix tree/hierarchy of nodes of content. Instead, you want to classify your documentation by different dimensions ("facettes") like in an online shop where you want your customers to search with multiple dimensions of the product (e.g. size, material, color). So, instead of one fix tree structure in your navigation panel, you want to have trees for each of your facette, e.g.:

- Size:
  - S
  - M
  - L
  - XL
- Color:
  - green
  - white
  - black
- Material:
  - Cotton
  - Leather
  - Plastic (although I can't imagine why one want to wear synthetic material, but that's another discussion ;-) )

If you click on one entry of a facette the rest of the facettes are filtered by that decision. Only those nodes and entries are displayed which meet the criterion indicated by the selected facette (or more than one criterion with AND-logical connection).

So, you don't have to decide at the beginning which tree structure for your content you want to have but you have only to classify your documentation topics according to the available facettes.

Later on, you could implement the possibility of dynamic hyperlinks which do not link to a specific note/site but to a classification (green leather shoes). Then, every note that meets these criteria gets linked (the link must be resolved into multiple links when creating the site).

**Comment by squidfunk** [comment]
This will be possible in the coming months with the new search and be part of the search interface, i.e., the search will support to categorize documents with arbitrary metadata. It will definitely not be part of the regular sidebar / navigation, since this is not the use case we are optimizing for. You can, however, [extend the theme](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme) and adjust the things that don't fit your use case ☺️

**Reply by AtrusRiven** [reply]
Thanks for your reply. Have you information on the new search feature or where to I have to look in order to get information on the new search feature? Thanks for your work!

**Reply by squidfunk** [reply]
Check the issue tracker, there are 7 open issues that we're going to address. Additionally, I'll open a new issue where I'll document progress once things are in a stable state. I'm making good progress, so likely in 2-3 weeks.

**Reply by squidfunk** [reply]
See #6307 for an update

**Comment by AtrusRiven** [comment]
Thanks a lot for your reply and your work!
